src/dsp/sample_analytics.cpp

```diff
@@ -54,19 +71,34 @@ namespace scxt::dsp {
         };
         }
     */
-        float ms = 0.0f;
-        const float divisor = static_cast<float>(s->channels) * static_cast<float>(s->getSampleLength());
-        for(size_t i = 0; i < s->getSampleLength(); i++) {
-            for(int chan = 0; chan < s->channels; chan++) {
-                ms += std::powf(*s->GetSamplePtrF32(chan), 2) / divisor;
+    float ms = 0.0f;
+    const float divisor =
+        static_cast<float>(s->channels) * static_cast<float>(s->getSampleLength());
+    for (size_t i = 0; i < s->getSampleLength(); i++)
+    {
+        for (int chan = 0; chan < s->channels; chan++)
+        {
+            float sample = 0.0;
+            switch (s->bitDepth)
+            {
+            case sample::Sample::BD_I16:
+                sample = static_cast<float>(*s->GetSamplePtrI16(chan)) / std::numeric_limits<int16_t>::max();
+                break;
+            case sample::Sample::BD_F32:
+                sample = *s->GetSamplePtrF32(chan);
+                break;
+                // what to do if a bitdepth isn't handled?
```

Comment:
Compiler warnings will kick in as errors if we add another storage flavor and don't handle it here.